### PR TITLE
Modify plan form when editing plans

### LINKED ISSUE #1378: RVL-1293 Thailand Preview and Local - Can't edit actions when plan saved as draft
On the manage Plans>Edit plan view
All fields should be editable on draft plans until the plans are activated, including being able to remove plan actions. Currently, if a plan is in draft, users can not edit specific fields and users can not remove actions.

### PR DESCRIPTION
**Pull request by ciremusyoka**
Closes #1378 
This PR enables hiding of some form field when editing plans and adding and removing of activities when editing draft case triggered plans as described [here](https://github.com/onaio/reveal-frontend/issues/1378#issuecomment-773263684). This functionalities are activated by `REACT_APP_HIDE_PLAN_FORM_FIELDS_ON_EDIT` and `React_APP_CASE_TRIGGERED_DRAFT_EDIT_ADD_ACTIVITIES` Envs respectively.

### CODE REVIEW

**Comment by ciremusyoka** [comment]
**Hidden fields**
draft_normal_plan

**Case triggered activities**
case_triggered

**Comment by moshthepitt** [comment]
@ciremusyoka how do you feel about having the list of fields to be hidden as the thing that is configurable, instead of having the intervention types as configurable but with static fields to be hidden?

**Comment by ciremusyoka** [comment]
> @ciremusyoka how do you feel about having the list of fields to be hidden as the thing that is configurable, instead of having the intervention types as configurable but with static fields to be hidden?

@moshthepitt  addressed this by https://github.com/onaio/reveal-frontend/pull/1537/commits/ac1d958b7177367bb79095bfcb05a44fca23c741

**Review by kelvin-muchiri — APPROVED** [review]

File: docs/env.md [hunk 1]
```diff
@@ -114,6 +114,17 @@ Below is a list of currently supported environment variables:
   - Optional parameter of plan intervention types whose tasks should be generated on opensrp
   - Options: FI, IRS, IRS_Lite, MDA, MDA-Point, Dynamic-FI, Dynamic-IRS and Dynamic-MDA
 
+- **REACT_APP_HIDE_PLAN_FORM_FIELDS_ON_EDIT**
+
+  - Optional parameter of plan intervention types whose some fields will be hidden when the plan status is draft, active or completed.
+  - Options: caseNum, interventionType, jurisdictions, fiReason, fiStatus, title, triggersAndConditions activityActionDefinitionUri, activityActionDescription, activityActionTitle, activityActionReason activityTimingPeriodStart, activityTimingPeriodEnd, activityGoalPriority, activityGoalValue
+
+- **React_APP_CASE_TRIGGERED_DRAFT_EDIT_ADD_ACTIVITIES**
```
Comment (moshthepitt) [inline docs/env.md:122]: Could you explain why we need to treat case triggered plans specially?

Comment (ciremusyoka) [reply docs/env.md:123]: Base on the request [here](https://github.com/onaio/reveal-frontend/issues/1378#issuecomment-773263684) adding and editing of the activities is only supposed to happen for case triggered plans only.

**Review by moshthepitt — DISMISSED** [review]

File: .env.sample [hunk 2]
```diff
@@ -177,4 +177,8 @@ REACT_APP_CHECK_SESSION_EXPIRY_STATUS=true
 REACT_APP_ASSIGNED_TEAMS_REQUEST_PAGE_SIZE=1000
 
 REACT_APP_OPENSRP_GENERATED_TASKS_INTERVENTIONS=Dynamic-FI,FI
+
+React_APP_CASE_TRIGGERED_DRAFT_EDIT_ADD_ACTIVITIES=false
```
[the remark this answers is not in the document]
Comment (ciremusyoka) [reply .env.sample]: corrected by https://github.com/onaio/reveal-frontend/pull/1537/commits/6fc2a1b19b29cede05136b9bf4930c792c1d9d5b

**Review by p-netm — APPROVED** [review]

**Review by kelvin-muchiri — APPROVED** [review]

**Pull request merged**